csrc/fusion_segmenter.cpp

```diff
@@ -4243,10 +4243,11 @@ void SegmentCandidateFinder::resolveNonscalarForwardedInput(
   SegmentedGroup* aux_group = input2group_.at(forwarded_input);
   NVF_ERROR(aux_group->producer_edges.empty());
 
-  std::vector<SegmentedGroup*> consumers;
+  // use unordered_set to avoid duplicated group in consumers.
+  std::unordered_set<SegmentedGroup*> consumers;
```

Comment:
How about VectorOfUniqueEntries instead? I'm worried that iterating a unordered_set is non-deterministic, which can lead to a serde problem like #1825. cc @rdspring1

Reply:
:bow: